src/App.js

```diff
@@ -1,25 +1,74 @@
-import logo from './logo.svg';
+import React, { useState } from 'react';
 import './App.css';
+import { Card, Row, Col, Divider, Input, Button } from 'antd';
+import foods from './foods.json';
+import FoodBox from './components/FoodBox';
+import AddFoodForm from './components/AddFoodForm';
+import Searchbar from './components/Searchbar';
 
 function App() {
+  const [food, setFood] = useState(foods);
+  const [alwaysFood, setAlwaysFood] = useState(foods);
+  const [visible, setVisible] = useState(true);
+  const [isFoodArrayEmpty, setIsFoodArrayEmpty] = useState(false);
+
+  const addToList = (item) => {
+    const foodCopy = [item, ...food];
+
+    setFood(foodCopy);
+  };
+
+  const searchFood = (query) => {
+    const filteredFood = food.filter((item) =>
+      item.name.toLowerCase().includes(query.toLowerCase())
+    );
+    if (filteredFood.length === 0) {
+      setIsFoodArrayEmpty(true);
+    }
+    setAlwaysFood(filteredFood);
+  };
+
+  const deleteFood = (foodName) => {
+    const filteredFood = food.filter((item) => {
+      return item.name !== foodName;
+    });
+    setFood(filteredFood);
+    setAlwaysFood(filteredFood);
+  };
+
+  const toggleVisibility = () => {
+    setVisible(!visible);
+  };
+
   return (
     <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+      <button onClick={toggleVisibility}>
+        {visible ? 'Hide Form' : 'Add New Food'}
+      </button>
+      {visible && <AddFoodForm addToList={addToList} />}
+      <Searchbar searchFood={searchFood} />
+
+      {isFoodArrayEmpty ? (
+        <p>oops ta fdd</p>
+      ) : (
```

Comment:
This message is not being displayed when there are no foods left. A easier way of doing this is by checking the `length` of the foods array, if it is empty display the message:
```
{!alwaysFood.length && <p>oops ta fdd</p>}
```